memo_app.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'sinatra/reloader'
+require 'cgi'
+require 'json'
+require 'securerandom'
+
+MEMOS_FILE = 'memos.json'
+
+not_found do
+  json(message: '404 Not Found.')
+end
+
+def load_memos
+  JSON.parse(File.read(MEMOS_FILE), symbolize_names: true)
+end
+
+def save_memos(indicated_data)
+  File.write(MEMOS_FILE, JSON.pretty_generate(indicated_data))
+end
+
+def find_memo(existing_data)
+  existing_data.find { |memo| memo[:id] == params[:id] }
+end
+
+existing_data = load_memos
+
+before do
+  @memos = existing_data
+end
+
+get '/' do
+  erb :index
+end
+
+get '/new' do
+  erb :new
+end
+
+post '/create' do
+  created_data = existing_data << { id: assign_id, title: params[:title], content: params[:content], created_at: Time.now.to_s, updated_at: Time.now.to_s }
+  save_memos(created_data)
+  redirect '/'
+end
+
+get '/:id' do
+  @memo = find_memo(existing_data)
+  erb :show
+end
+
+get '/:id/edit' do
+  @memo = find_memo(existing_data)
+  erb :edit
+end
+
+patch '/:id/update' do
+  @memo = find_memo(existing_data)
+  @memo.merge!({ title: params[:title], content: params[:content], updated_at: Time.now.to_s })
+  updated_data = existing_data
+  save_memos(updated_data)
```

Comment:
```suggestion
  save_memos(existing_data)
```

merge! したのなら、元のデータがそのまま書き換わっているはずです。
チェリー本の2章で説明している参照の概念を読み直してみてください。

Reply:
```save_memos```に渡すデータの名前を変更すべきか検討してました。updateのアクションを終えたデータなので、```existing_data```よりも```updated_data```の方が分かりやすいかなと考えました。わざわざ、変数名をへんこうするなら、そのままの変数名でよい、という考えの方がベターでしょうか？

Reply:
そうですね。
変数名を変えた方がわかりやすくなるケースがゼロとは言いませんが、僕は滅多にやらないですね。
そもそも、

> updateのアクションを終えたデータなので、existing_dataよりもupdated_dataの方が分かりやすい

と感じるなら、最初に命名した `existing_data` という変数名に問題がある、と考えた方が良いかもしれません。

参考 https://bootcamp.fjord.jp/articles/78